src/jit/codegen.h

```diff
@@ -548,13 +548,13 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
     //-------------------------------------------------------------------------
     // scope info for the variables
 
-    void genSetScopeInfo(unsigned        which,
-                         UNATIVE_OFFSET  startOffs,
-                         UNATIVE_OFFSET  length,
-                         unsigned        varNum,
-                         unsigned        LVnum,
-                         bool            avail,
-                         const siVarLoc* varLoc);
+    void genSetScopeInfo(unsigned       which,
+                         UNATIVE_OFFSET startOffs,
+                         UNATIVE_OFFSET length,
+                         unsigned       varNum,
+                         unsigned       LVnum,
+                         bool           avail,
+                         siVarLoc&      varLoc);
```

Comment:
Why was this changed back to non-const reference? We discussed in the past that such function arguments are not recommended.

Reply:
https://github.com/dotnet/coreclr/pull/23373#discussion_r269344497

Reply:
I don't see anything in that comment about pointer vs. reference, it's only about being const or not. If it has to be non-const then it should be a pointer rather than a reference.

Reply:
"varLoc" should never be a nullptr when reporting a variable location. Why would we let that happen? The other option is to add an
> assert( siVarLoc != nullptr)

but it would be the same.

Reply:
I don't see why you need an assert for that. There are many functions in the JIT that take pointers and do not have such asserts. In general a null pointer assert is only necessary if the function does not dereference the pointer and instead stores it somewhere for later use.

Reply:
Okey, and in this case we would fail inside the method when we try to do the copy by value in eeVars. Now it is just a preference between failing on the call or failing when are trying to copy it. It is the same for me. Is there any specific reason for pointer over reference?

Reply:
> Now it is just a preference between failing on the call or failing when are trying to copy it.

I can't speak for the rest of the team but I doubt anyone would have any concerns about this. There are simply too few uses of this function for this to matter. Nobody's going to pass nullptr to it.

> Is there any specific reason for pointer over reference?

The problem with non-const reference arguments is that it makes the code less readable. If you have:
```C++
Foo foo;
bar(foo);
```
it's not readily obvious that `bar` may modify `foo`. Using pointers provides a hint:
```C++
Foo foo;
bar(&foo);
```
See JIT coding guidelines: https://github.com/dotnet/coreclr/blob/master/Documentation/coding-guidelines/clr-jit-coding-conventions.md#15.3.5

Reply:
Thanks!